tests/util/mod.rs

```diff
@@ -25,19 +25,148 @@ use regex::bytes::Regex;
+    pub async fn create_temporary_table(
+        &mut self,
+        pk: &'static str,
+        sk: Option<&'static str>,
+    ) -> Result<String, Box<dyn std::error::Error>> {
+        let table_name: String = rand::thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(16)
+            .map(char::from)
+            .collect();
+        println!("create temporary table: {}", table_name);
+
+        let mut c = self.command()?;
+        let port = self.port.to_string();
+        let mut args = vec![
+            "--region",
+            "local",
+            "--port",
+            &port,
+            "admin",
+            "create",
+            "table",
+            &table_name,
+            "--keys",
+        ];
+        let mut keys = vec![pk];
+        if let Some(sk) = sk {
+            keys.push(sk);
+        }
+
+        args.extend(keys);
+        c.args(args).assert().success();
+
+        self.temporary_tables.push(table_name.clone());
+
+        Ok(table_name)
+    }
+
+    /// Create temporary table with items via `create_temporary_table`.
+    pub async fn create_temporary_table_with_items(
+        &mut self,
+        pk: &'static str,
+        sk: Option<&'static str>,
+        items: Vec<TemporaryItem>,
+    ) -> Result<String, Box<dyn std::error::Error>> {
+        let table_name = self.create_temporary_table(pk, sk).await?;
+
+        for ti in items {
+            let mut c = self.command()?;
+            let mut args = vec!["--region", "local", "--table", &table_name, "put"];
+            args.extend(ti.keys());
+            if let Some(item) = ti.item {
+                args.extend(vec!["--item", item]);
+            }
+
+            c.args(args).assert().success();
+        }
+
+        Ok(table_name)
+    }
+
+    /// Delete table manually.
+    pub fn cleanup<I, S>(&self, tables: I) -> Result<(), Box<dyn std::error::Error>>
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<str>,
+    {
+        for table in tables {
+            let mut c = self.command()?;
+            let cmd = c.args(&[
+                "--region",
+                "local",
+                "--port",
+                &self.port.to_string(),
+                "admin",
+                "delete",
+                "table",
+                "--yes",
+                table.as_ref(),
+            ]);
+            cmd.assert().success();
+        }
+
+        Ok(())
+    }
+}
+
+impl<'a> Drop for TestManager<'a> {
+    fn drop(&mut self) {
+        let _ = self.cleanup(&self.temporary_tables);
+    }
+}
+
+pub async fn setup<'a>() -> Result<TestManager<'a>, Box<dyn std::error::Error>> {
     setup_with_port(8000).await
 }
 
-// We use std::sync::Mutex instead of tokio::sync::Mutex, because mutex must be poisoned after setup failure.
-static SETUP_MUTEX: Lazy<Mutex<i32>> = Lazy::new(|| Mutex::new(0));
+pub async fn setup_with_port<'a>(port: i32) -> Result<TestManager<'a>, Box<dyn std::error::Error>> {
```

Comment:
Because `TestManager` is created inside of the function, we can use `pub async fn setup_with_port(port: i32) -> Result<TestManager<'static>, Box<dyn std::error::Error>>`.